packages/core/admin/admin/src/content-manager/pages/EditSettingsView/components/GenericInput.tsx

```diff
@@ -1,23 +1,36 @@
-import React from 'react';
-
 import { Option, Select, TextInput, ToggleInput } from '@strapi/design-system';
-import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
 
-const GenericInput = ({ type, options, onChange, value, name, ...inputProps }) => {
+interface GenericInputProps extends Record<string, unknown> {
+  type: string;
+  options?: string[];
+  onChange: (e: { target: { name: string; value: string | boolean | number } }) => void;
+  value: string | boolean | string[];
+  name: string;
+  label: string;
+}
+
+const GenericInput = ({
```

Comment:
```suggestion
/**
 * TODO: refactor this component because it's not
 * the best way to do modular rendering, hence why
 * we're casting all the values for example.
 */
const GenericInput = ({
```